app/controllers/tasks.js

```diff
@@ -263,15 +277,71 @@ export default class TasksController extends Controller {
     }
   }
 
-  @action async handleUpdateTask(taskId, error) {
+  @action async handleUpdateTask(taskId, resetCurrentTask) {
+    this.resetCurrentTask = resetCurrentTask;
     const taskData = this.taskFields;
-    if (taskData.percentCompleted === TASK_PERCENTAGE.completedPercentage) {
+    const currentTask = this.getTaskById(taskId);
+    const isCurrentTaskStatusInProgress =
+      currentTask.status === this.TASK_KEYS.IN_PROGRESS;
+    const isCurrentTaskStatusBlock =
+      currentTask.status === this.TASK_KEYS.BLOCKED;
+    const isNewTaskStatusInProgress =
+      taskData.status === this.TASK_KEYS.IN_PROGRESS;
+    const isNewTaskStatusBlock = taskData.status === this.TASK_KEYS.BLOCKED;
+    const isCurrProgress100 =
+      parseInt(currentTask.percentCompleted || 0) === 100;
+    const isCurrProgress0 = parseInt(currentTask.percentCompleted || 0) === 0;
+
+    if (this.dev && taskData.status) {
+      if (
+        isNewTaskStatusInProgress &&
+        !isCurrentTaskStatusBlock &&
+        !isCurrProgress0
+      ) {
+        this.message = 'Proceeding further will make task progress 0%.';
+        this.showModal = true;
+        this.buttonRequired = true;
+        this.tempTaskId = taskId;
+        this.taskFields.percentCompleted = 0;
+
+        return;
+      }
+      if (
+        isCurrentTaskStatusInProgress &&
+        !isNewTaskStatusBlock &&
+        !isCurrProgress100
+      ) {
+        this.message = 'Proceeding further will make task progress 100%.';
+        this.showModal = true;
+        this.buttonRequired = true;
+        this.tempTaskId = taskId;
+        this.taskFields.percentCompleted = 100;
+        return;
+      }
+      if (
+        isCurrentTaskStatusBlock &&
+        !isNewTaskStatusInProgress &&
+        !isCurrProgress100
```

Comment:
Move !isCurrProgress100 to the parent if, I see it is false for all the conditions

Reply:
Done